scipy/integrate/_quadrature.py

```diff
@@ -1035,3 +1038,235 @@ def newton_cotes(rn, equal=0):
+                   "including the boundaries of the integration range; "
+                   "e.g. `func(*ranges[:, 0])` and `func(*ranges[:, 1])` "
+                   "must return the integrand at the lower and upper "
+                   "integration limits.")
+        raise ValueError(message) from e
+
+    try:
+        func(*ranges, *args)
+        vfunc = func
+    except Exception as e:
+        wmessage = ("Exception encountered when attempting vectorized call to "
+                    f"`func`: {e}. `func` should accept arrays `x0, ..., xn` "
+                    "for better performance.")
+        emessage = ("Exception encountered when attempting vectorized call to "
+                    f"`func`: {e}, and `args` is not `None`. `func` must "
+                    "accept arrays `x0, ..., xn` if `args` is not `None`.")
+        if args != tuple():
+            raise ValueError(emessage)
+        else:
+            warnings.warn(wmessage)
+        vfunc = np.vectorize(func)
+
+    n_points_int = np.int64(n_points)
+    if n_points != n_points_int:
+        message = "`n_points` must be an integer."
+        raise TypeError(message)
+
+    n_offsets_int = np.int64(n_offsets)
+    if n_offsets != n_offsets_int:
+        message = "`n_offsets` must be an integer."
+        raise TypeError(message)
+
+    if qrng is None:
+        qrng = stats.qmc.Halton(dim)
+    elif not isinstance(qrng, stats.qmc.QMCEngine):
+        message = "`qrng` must be an instance of scipy.stats.qmc.QMCEngine."
+        raise TypeError(message)
+
+    if qrng.d != ranges.shape[0]:
+        message = ("`qrng` must be initialized with dimensionality equal to "
+                   "the number of variables in `ranges`, i.e., "
+                   "`qrng.random().shape[-1]` must equal `ranges.shape[0]`.")
+        raise ValueError(message)
+
+    rng_seed = getattr(qrng, 'rng_seed', None)
+    rng = stats._qmc.check_random_state(rng_seed)
+
+    if log not in {True, False}:
+        message = "`log` must be boolean (`True` or `False`)."
+        raise TypeError(message)
+
+    return (vfunc, ranges, n_points_int, n_offsets_int,
+            qrng, rng, log, args, stats)
+
+
+QMCQuadResult = namedtuple('QMCQuadResult', ['integral', 'standard_error'])
+
+
+def qmc_quad(func, ranges, *, n_points=1024, n_offsets=8, qrng=None, log=False,
+             args=None):
+    """
+    Compute an integral in N-dimensions using Quasi-Monte Carlo quadrature.
+
+    Parameters
+    ----------
+    func : callable
+        The integrand. Must accept separate arguments ``x0, ..., xn``
+        corresponding with the coordinate in each dimension. For efficiency,
+        the function should be vectorized to compute the integrand for each
+        element of array inputs ``x0, ..., xn``, each of length `n_points`.
+        If the function is vectorized, it may also accept additional positional
+        arguments specified with `args`.
+    ranges : array-like
```

Comment:
I would name this `limits` instead, or something else, as it's close to `range` which could be confusing.

Reply:
Done - well, `a`, and `b`